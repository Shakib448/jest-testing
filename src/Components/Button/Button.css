.button__btn {
  border: 1px solid gray;
  padding: 10px;
}
