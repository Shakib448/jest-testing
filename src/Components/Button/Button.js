import React from "react";
import "./Button.css";
const Button = ({ label }) => {
  return (
    <div data-testid="button" className="button__btn">
      {label}
    </div>
  );
};
export default Button;
