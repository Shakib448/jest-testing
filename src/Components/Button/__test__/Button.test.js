import React from "react";
import ReactDOM from "react-dom";
import Button from "../Button";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button label="save"></Button>);
  expect(getByTestId("button")).toHaveTextContent("save");
});

it("match snapshot", () => {
  const tree = renderer.create(<Button label="save"></Button>);
  expect(tree).toMatchSnapshot();
});
it("match snapshot 2", () => {
  const tree = renderer.create(<Button label="save"></Button>);
  expect(tree).toMatchSnapshot();
});
it("match snapshot 3", () => {
  const tree = renderer.create(<Button label="save"></Button>);
  expect(tree).toMatchSnapshot();
});
