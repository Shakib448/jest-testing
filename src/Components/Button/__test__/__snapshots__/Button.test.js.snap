// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mathces snapshot 1`] = `
<div
  className="button__btn"
  data-testid="button"
>
  save
</div>
`;
